### PR TITLE
features: antiaffinity/remove tolerations/remove elaticsearch crds/sa…

### PR DESCRIPTION
**Pull request by wangzhuzhen**
…ve pvc for cassandra

#### What type of this PR
/kind feature

#### What this PR does / why we need it:
1. Pod for components ES, Etcd, Kafka/Zookeeper, Cassandra, Redis set hard pod anti-affinity for high available deployment
2. save pvc for cassandra pods
3. delete unnecessary crds for  elasticsearch
4. kafka pods and zookeeper pods tight coupled

#### Which issue(s) this PR fixes:
Fix part of issue 46(https://github.com/erda-project/erda-release/issues/46) 

#### Specified Reviewers:

/assign @iutx @luobily 

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: erda-helm/erda/charts/elasticsearch/templates/statefulset.yaml [hunk 1]
```diff
@@ -21,9 +22,28 @@ spec:
...
+                operator: Exists
+        podAntiAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+          - labelSelector:
+              matchExpressions:
+              - key: componentName
+                operator: In
+                values:
+                - erda-elasticsearch
+            topologyKey: "kubernetes.io/hostname"
```
Comment (luobily) [inline erda-helm/erda/charts/elasticsearch/templates/statefulset.yaml:46]: Does this requirement also exist in demo mode?

File: erda-helm/erda/charts/etcd/templates/etcd-cluster.yaml [hunk 2]
```diff
@@ -38,9 +39,28 @@ spec:
...
+                  operator: Exists
+        podAntiAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            - labelSelector:
+                matchExpressions:
+                - key: componentName
+                  operator: In
+                  values:
+                  - erda-etcd
+              topologyKey: "kubernetes.io/hostname"
```
Comment (luobily) [inline erda-helm/erda/charts/etcd/templates/etcd-cluster.yaml:63]: Does this requirement also exist in demo mode?

File: erda-helm/erda/charts/etcd/templates/etcd-cluster.yaml [hunk 3]
```diff
@@ -158,9 +179,28 @@ spec:
...
```
Comment (luobily) [inline erda-helm/erda/charts/etcd/templates/etcd-cluster.yaml:203]: Does this requirement also exist in demo mode?

File: erda-helm/erda/charts/etcd/templates/etcd-cluster.yaml [hunk 4]
```diff
@@ -278,9 +319,28 @@ spec:
...
+                  operator: Exists
+        podAntiAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            - labelSelector:
+                matchExpressions:
+                  - key: componentName
+                    operator: In
+                    values:
+                    - erda-etcd
+              topologyKey: "kubernetes.io/hostname"
```
Comment (luobily) [inline erda-helm/erda/charts/etcd/templates/etcd-cluster.yaml:343]: Does this requirement also exist in demo mode?

File: erda-helm/erda/charts/kafka/templates/cluster.yaml [hunk 5]
```diff
@@ -19,20 +20,39 @@ spec:
...
+                topologyKey: "kubernetes.io/hostname"
+        podAntiAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            - labelSelector:
+                matchExpressions:
+                  - key: componentName
+                    operator: In
+                    values:
+                    - erda-kafka
+              topologyKey: "kubernetes.io/hostname"
```
Comment (luobily) [inline erda-helm/erda/charts/kafka/templates/cluster.yaml:55]: +1

File: erda-helm/erda/charts/zookeeper/templates/cluster.yaml [hunk 6]
```diff
@@ -138,20 +148,28 @@ spec:
...
+                  operator: Exists
+        podAntiAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            - labelSelector:
+                matchExpressions:
+                - key: componentName
+                  operator: In
+                  values:
+                  - erda-zookeeper
+              topologyKey: "kubernetes.io/hostname"
```
Comment (luobily) [inline erda-helm/erda/charts/zookeeper/templates/cluster.yaml:172]: +1

File: erda-helm/erda/charts/zookeeper/templates/cluster.yaml [hunk 7]
```diff
@@ -257,20 +276,28 @@ spec:
...
```
Comment (luobily) [inline erda-helm/erda/charts/zookeeper/templates/cluster.yaml:300]: +1

File: erda-helm/erda/charts/elasticsearch/templates/statefulset.yaml (hunk 1 above)
Comment (wangzhuzhen) [reply erda-helm/erda/charts/elasticsearch/templates/statefulset.yaml:45]: Have fixed.

File: erda-helm/erda/charts/etcd/templates/etcd-cluster.yaml (hunk 3 above)
Comment (wangzhuzhen) [reply erda-helm/erda/charts/etcd/templates/etcd-cluster.yaml:199]: No, demo mode use etcd-demo.yaml file.

File: erda-helm/erda/charts/etcd/templates/etcd-cluster.yaml (hunk 4 above)
Comment (wangzhuzhen) [reply erda-helm/erda/charts/etcd/templates/etcd-cluster.yaml:337]: No, demo mode use etcd-demo.yaml file.

File: erda-helm/erda/charts/kafka/templates/cluster.yaml (hunk 5 above)
Comment (wangzhuzhen) [reply erda-helm/erda/charts/kafka/templates/cluster.yaml:53]: No, demo mode use demo.yaml file.

File: erda-helm/erda/charts/zookeeper/templates/cluster.yaml (hunk 6 above)
Comment (wangzhuzhen) [reply erda-helm/erda/charts/zookeeper/templates/cluster.yaml:168]: No, demo mode use demo.yaml file.

File: erda-helm/erda/charts/zookeeper/templates/cluster.yaml (hunk 7 above)
Comment (wangzhuzhen) [reply erda-helm/erda/charts/zookeeper/templates/cluster.yaml:294]: No, demo mode use demo.yaml file.

**Pull request merged**